source/components/hardware/hwregs.c

```diff
@@ -167,13 +167,21 @@ AcpiHwGetAccessBitWidth (
 
     if (!Reg->AccessWidth)
     {
-        if (Reg->SpaceId == ACPI_ADR_SPACE_SYSTEM_IO)
+        if (Reg->BitWidth && !(Reg->BitWidth & 0x07) &&
+            ACPI_IS_POWER_OF_TWO (Reg->BitWidth >> 3))
         {
```

Comment:
It looks we could further limit this to:
if (!Reg->BitOffset && Reg->BitWidth &&
    ACPI_IS_POWER_OF_TWO (Reg->BitWidth) && ACPI_IS_ALIGNED(Reg->BitWidth, 8) &&
    ACPI_IS_ALIGNED(Reg->Address))